int32 goal_id
int32 x
int32 y